lib/schema-fs-loader.js

```diff
@@ -84,19 +132,40 @@ const load = (dir, context, callback) => {
 //     error - <Error>
 //     metaschema - <Metaschema>
 const loadAndCreate = (dir, context, callback) => {
-  if (Array.isArray(dir)) {
-    metasync.map(dir,
-      (d, cb) => { load(d, context, cb); },
-      (err, schemas) => {
-        if (err) callback(err);
-        else callback(...create([].concat(...schemas)));
+  const schemas = {
+    categories: [],
+    domains: [],
+    views: [],
+    forms: [],
+    actions: [],
+    sources: [],
+  };
+
+  const finish = err => {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    loadModuleSchema(domainsPath, context, schemas, schemasDir, err => {
+      if (err) {
+        callback(err);
+      } else {
+        callback(...create(schemas));
       }
+    });
+  };
+
+  if (Array.isArray(dir)) {
+    metasync.map(
```

Comment:
Why use `metasync.map()` here, if you are not using the resulting array in the callback afterward?
```suggestion
    metasync.each(
```